### ISSUE TITLE
We need to clean up test files better

### ISSUE THREAD

**Issue by Dead2** [zlib-ng/zlib-ng#728]
Currently `git status` tells me I have the following untracked files:

```
        test/GH-382/defneg3.dat.out
        test/GH-382/defneg3.dat.zz
        test/data/fireworks.jpg.gz
        test/data/fireworks.jpg.gzip.gz
        test/data/fireworks.jpg.gzip.out
        test/data/fireworks.jpg.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h-minigzip-lcet10-zl-5F.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h-minigzip-lcet10-zl-5F.gzip.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h-minigzip-lcet10-zl-5F.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h.gzip.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h.gzip.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6-minigzip-lcet10-zl-h.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6.gzip.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0-minigzip-lcet10-zl-6.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0.gzip.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0-minigzip-lcet10-zl-0.out
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6-minigzip-lcet10-zl-0.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f-minigzip-lcet10-zl-6.gz
        test/data/lcet10-zl.txt-minigzip-lcet10-zl-9f.gz
        test/data/lcet10-zl.txt.gz
        test/data/lcet10.txt.gz
        test/data/lcet10.txt.gzip.gz
        test/data/lcet10.txt.gzip.out
        test/data/lcet10.txt.out
        test/data/paper-100k.pdf.gz
        test/data/paper-100k.pdf.gzip.gz
        test/data/paper-100k.pdf.gzip.out
        test/data/paper-100k.pdf.out
```

There might also be some that gets ignored due to `.gitignore`.
Some of these might be old leftovers though, but I think I cleaned them out a few weeks ago.

We should get better at deleting the files after the test or with `make clean`.
The repeating file pattern above also hints that we might have (or had) a bug with selecting files for testing.

Maybe we could also put temp files in `test/tmp` perhaps, making them easier to separate from legitimate files and easier to clean up manually if need be.

**Comment by nmoinvaz** [comment]
I think perhaps the long filename issue is partially resolved by #732 by adding quotes around `TEST_FILE_PATH`.

**Comment by nmoinvaz** [comment]
I think we can close this now.
